src/AccordionCollapse.js

```diff
@@ -0,0 +1,37 @@
+import classNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Collapse from './Collapse';
+import { createBootstrapComponent } from './ThemeProvider';
+import AccordionContext from './AccordionContext';
+
+class AccordionCollapse extends React.Component {
+  static propTypes = {
+    /** @default 'accordion-collapse' */
+    bsPrefix: PropTypes.string,
+    /**
+     * A key that corresponds to the toggler that triggers this collapse's expand or collapse.
+     */
+    eventKey: PropTypes.string.isRequired,
+  };
+
+  render() {
+    const { bsPrefix, children, className, eventKey, ...props } = this.props;
+
+    return (
+      <AccordionContext.Consumer>
+        {context => (
+          <Collapse in={context.activeEventKey === eventKey} {...props}>
```

Comment:
this kind of pattern has historically been problematic for us in the past. e.g. we're not passing `style` down to the wrapped child

Reply:
In this case Context is just passing props through to the `div` and rendering nothing. Agree it's not great tho. It's more of a convenience to not have to split out transition specific props, but we should have better pattern for that.

Reply:
is there a better pattern I should look into? I'm not sure as to what other approach I should go with

Reply:
I think this ok. It's the underlying Collapse component that's the problem